optuna/samplers/_gp/_sampler.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import annotations
+
+from typing import Any
+from typing import Callable
+from typing import Sequence
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+import optuna
+from optuna._experimental import experimental_class
+from optuna._imports import _LazyImport
+from optuna.distributions import BaseDistribution
+from optuna.samplers._base import BaseSampler
+from optuna.study import Study
+from optuna.study import StudyDirection
+from optuna.trial import FrozenTrial
+from optuna.trial import TrialState
+
+
+if TYPE_CHECKING:
+    import torch
+
+    import optuna._gp._acqf as _acqf
+    import optuna._gp._gp as _gp
+    import optuna._gp._optim as _optim
+    import optuna._gp._search_space as _search_space
+else:
+    torch = _LazyImport("torch")
+    _search_space = _LazyImport("optuna._gp._search_space")
+    _gp = _LazyImport("optuna._gp._gp")
+    _optim = _LazyImport("optuna._gp._optim")
+    _acqf = _LazyImport("optuna._gp._acqf")
+
+
+def default_log_prior(kernel_params: "_gp.KernelParams") -> "torch.Tensor":
+    def gamma_log_prior(x: "torch.Tensor", concentration: float, rate: float) -> "torch.Tensor":
+        return (concentration - 1) * torch.log(x) - rate * x
+
+    return (
+        gamma_log_prior(kernel_params.inv_sq_lengthscales, 2, 0.5).sum()
+        + gamma_log_prior(kernel_params.kernel_scale, 2, 1)
+        + gamma_log_prior(kernel_params.noise, 1.1, 20)
+    )
+
+
+@experimental_class("3.6.0")
+class GPSampler(BaseSampler):
+    """Sampler using Gaussian process-based Bayesian optimization.
+
+    This sampler fits a Gaussian process (GP) to the objective function and optimizes
+    the acquisition function to suggest the next parameters.
+
+    The current implementation uses:
+    - Matern kernel with nu=2.5 (differentiable twice)
+    - Automatic relevance determination (ARD) for a lengthscale of each parameter
+    - Gamma prior for lengthscales, kernel scale, and noise scale
+    - Log Expected Improvement (logEI) as the acquisition function
+    - Quasi-Monte Carlo (QMC) sampling to optimize the acquisition function
```

Comment:
[nit] I changed it with the Oxford comma.
```suggestion
    The current implementation uses:
        - Matern kernel with nu=2.5 (twice differentiable),
        - Automatic relevance determination (ARD) for the length scale of each parameter,
        - Gamma prior for lengthscales, kernel scale, and noise scale,
        - Log Expected Improvement (logEI) as the acquisition function, and
        - Quasi-Monte Carlo (QMC) sampling to optimize the acquisition function.
```